contracts/src/xchain.ts

```diff
@@ -0,0 +1,415 @@
+import { Contract, JsonRpcProvider, toBeHex, ZeroHash, solidityPackedKeccak256,
+         zeroPadValue, formatUnits, encodeRlp, decodeRlp, BytesLike, hexlify
+} from "ethers"
+
+import { GetProofResponse, TokenInfo } from "./types.js";
+import { Block, BlockOptions, JsonRpcBlock } from "@ethereumjs/block";
+import { Common, CustomChain } from "@ethereumjs/common";
+
+export function randomchoice<T>(array:T[]):T {
+  return array[Math.floor(Math.random() * array.length)];
+}
+
+export const chain_info: Record<number,any> = {
```

Comment:
This list was probably combined from the ethereum-lists/chains repo?

```suggestion
// A subset of chains (see https://github.com/ethereum-lists/chains/tree/master/_data/chains).
export const chain_info: Record<number,any> = {
```